dresscode/corrections.py

```diff
@@ -400,17 +398,12 @@ def coicorr_uncert_cts(
     ):
         header = primary_frame.header
 
-        coi_loss_corr_unc_cts = primary_frame.data * corr_factor_rel_unc_frame.data
-        coi_loss_corr_unc_cts[np.isnan(coi_loss_corr_unc_cts)] = 0
-
-        # todo: square before summing
-        # squared_coi_loss_corr_unc_cts = (
-        #     primary_frame.data * corr_factor_rel_unc_frame.data
-        # ) ** 2
-        # squared_coi_loss_corr_unc_cts[np.isnan(squared_coi_loss_corr_unc_cts)] = 0
-        # new_hdu = fits.ImageHDU(squared_coi_loss_corr_unc_cts, header)
+        coi_loss_corr_unc_cts_squared = np.square(
+            primary_frame.data * corr_factor_rel_unc_frame.data
+        )
```

Comment:
@mdecleir just want to be clear I am doing what we had discussed. we want to convert the uncertainty into counts here (by multiplying the uncertainty by the primary frame) and then we want to square the counts?

The alternative is to possibly square the uncertainty first, and _then_ convert to counts?

Reply:
This is correct. We are summing the primary frames in counts, so we should be quadratically summing the uncertainties in counts as well.

Reply:
I am not sure though if the squaring should be happening in the "corrections.py" step. We only need to square when summing images, so I think it would be more appropriate to do this right before the summing step, if possible.

Reply:
> I am not sure though if the squaring should be happening in the "corrections.py" step. We only need to square when summing images, so I think it would be more appropriate to do this right before the summing step, if possible.

Because the files are already open, in memory, I put the conversion to counts inside `corrections.py` so that we can just operate on the files in memory. My internal justification of this was that corrections.py took counts as inputs, and it spits out counts as outputs.

However, I can see how it might make more sense inside uvotimsum.py. We'd lose a little efficiency in processing the files since we have to go to the file system an extra time to load them, but it's not a big deal.  Mostly would be just moving some code around. I'm happy to do whichever you prefer.

Reply:
But in uvotimsum you have to open the files anyway, so doing the quadratic sum there after opening them, shouldn't make a difference right? That being said, I have not checked the code in detail, so at this point, I might have lost track of what files are being created and opened where.

In any case, I don't think it is necessary to write out files with the uncertainties in quadrature. That just adds to the amount of data being written.

Reply:
Yes, I think there's a way to avoid an extra file read in uvotimsum if we did this operation as we append frames to an "all" file, but then that section of the pipeline gets more complicated because it's not just doing an append, it's also worrying about what type of file it is, and converting to counts, and sometimes squaring the data. 

I think it's better to leave as is for now, at least with my current time constraint, as there are tradeoffs that need to be balanced. If we want to refactor this later, I think that's fine and makes sense.

Reply:
Ok, makes sense to me. If you can, just make sure that the header keyword that gets saved with it is correct. This avoids confusing in the future.